src/DynamoCore/Extensions/ExtensionManager.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Dynamo.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Dynamo.Extensions
+{
+    public class ExtensionManager: IExtensionManager, ILogSource
+    {
+        private readonly List<IExtension> extensions = new List<IExtension>();
+        private readonly ExtensionLoader extensionLoader = new ExtensionLoader();
+
+        public ExtensionManager()
+        {
+            extensionLoader.MessageLogged += Log;
+        }
+
+        public void Add(IExtension extension)
+        {
+            if (extensions.Find(ext => ext.UniqueId == extension.UniqueId) == null)
+            {
+                extensions.Add(extension);
+                Log(extension.Name + " extension is added");
+
+                if (ExtensionAdded != null)
+                {
+                    ExtensionAdded(extension);
+                }
+            }
+            else
+            {
+                Log("Could not add a duplicate of " + extension.Name);
+            }
+        }
+
+        public void Remove(IExtension extension)
+        {
+            if (!extensions.Contains(extension))
+            {
+                Log("ExtensionManager does not contain a specified extension");
+                return;
```

Comment:
This is a minor issue, but let's make sure we note the name and guid of the extension here.